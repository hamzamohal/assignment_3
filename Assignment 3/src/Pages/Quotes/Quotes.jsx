import './quotes.scss';
import quotes from '../../Data/quotes';

const Quotes = () => {
  const takeRandomQuote = (quotes, start = 0) => {
    const index = Math.floor(Math.random() * (quotes.length - start + 1) + start);
    return quotes[index];
  };

  const quote = takeRandomQuote(quotes).split('-');

  return (
    <div className="quotes">
      <blockquote>{quote[1]}</blockquote>
      <p>{quote[0]}</p>
    </div>
  );
};

export default Quotes;
