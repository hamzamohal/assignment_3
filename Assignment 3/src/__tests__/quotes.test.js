import TestRenderer from 'react-test-renderer';
import Quotes from '../Pages/Quotes/Quotes';

describe('Test Quotes component', () => {
  beforeEach(() => {
    jest.spyOn(Math, 'random').mockReturnValue(0.86);
  });

  it('should render Quotes component', () => {
    const quoteComponent = TestRenderer.create(<Quotes />).toJSON();
    expect(quoteComponent).toMatchSnapshot();
  });
});
