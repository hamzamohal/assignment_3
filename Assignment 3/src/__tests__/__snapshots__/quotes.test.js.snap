// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Quotes component should render Quotes component 1`] = `
<div
  className="quotes"
>
  <blockquote>
    If you stop at general math, then you will only make general money
  </blockquote>
  <p>
    Snoop Dogg
  </p>
</div>
`;
